Client/Frontend/Browser/LoginsHelper.swift

```diff
@@ -180,48 +179,185 @@ class LoginsHelper: TabContentScript {
+                self.tab?.removeSnackbar(bar)
+                self.snackBar = nil
                 return
+        }
+        
+        let saveORUpdate = SnackButton(
+            title: isUpdating ?  Strings.loginsHelperUpdateButtonTitle : Strings.loginsHelperSaveLoginButtonTitle,
+            accessibilityIdentifier: "UpdateLoginPrompt.saveUpdateButton") { [unowned self] bar in
+                self.tab?.removeSnackbar(bar)
+                self.snackBar = nil
+                
+                completion()
+        }
+        
+        snackBar?.addButton(dontSaveORUpdate)
+        snackBar?.addButton(saveORUpdate)
+        
+        tab?.addSnackbar(snackBar)
+    }
+    
+    private func autoFillRequestedCredentials(formSubmitURL: String, logins: [PasswordForm], requestId: String, frameInfo: WKFrameInfo) {
+        let securityOrigin = frameInfo.securityOrigin
+        
+        var jsonObj = [String: Any]()
+        jsonObj["requestId"] = requestId
+        jsonObj["name"] = "RemoteLogins:loginsFound"
+        jsonObj["logins"] = logins.compactMap { loginData -> [String: String]? in
+            if frameInfo.isMainFrame {
+                return loginData.toDict(formSubmitURL: formSubmitURL)
             }
             
-            self.tab?.webView?.evaluateSafeJavaScript(functionName: "window.__firefox__.logins.inject", args: [jsonString], contentWorld: .defaultClient, escapeArgs: false) { (obj, err) -> Void in
-                if err != nil {
-                    log.debug(err)
-                }
+            // Check for current tab has a url to begin with
+            // and the frame is not modified
+            guard let currentURL = tab?.webView?.url,
+                  LoginsHelper.checkIsSameFrame(
+                    url: currentURL,
+                    frameScheme: securityOrigin.protocol,
+                    frameHost: securityOrigin.host,
+                    framePort: securityOrigin.port) else {
+                return nil
+            }
+            
+            // If it is not the main frame, return username only, but no password!
+            // Chromium does the same on iOS.
+            // Firefox does NOT support third-party frames or iFrames.
+            if let updatedLogin = loginData.copy() as? PasswordForm {
+                updatedLogin.update(loginData.usernameValue, passwordValue: "")
+                
+                return updatedLogin.toDict(formSubmitURL: formSubmitURL)
             }
+            
+            return nil
         }
+        
+        let json = JSON(jsonObj)
+        guard let jsonString = json.stringValue() else {
+            return
+        }
+        
+        self.tab?.webView?.evaluateSafeJavaScript(
+            functionName: "window.__firefox__.logins.inject",
+            args: [jsonString],
+            contentWorld: .defaultClient,
+            escapeArgs: false) { (obj, err) -> Void in
+            if err != nil {
+                log.debug(err)
```

Comment:
Would we want to do `log.error` here?

Reply:
Fix at https://github.com/brave/brave-ios/pull/5013/commits/1a6d7cfd93a42bc6c37576fe5ddbca88aba7fa7c